### ISSUE TITLE
Support for games without option to connect trough IP + command line parametters

### ISSUE THREAD

**Issue by M4RCK5** [gyf304/partylan#4]
Hi @gyf304 , nice tool u have here.

Ive seen that ur tool creates a virtual network adapter (like Zerotier/Radmin/Hamachi, etc.), i was thinking if it could work too with games that dont have the option to connect trough IP, ive tried it too but doesnt work for those, can we have an option to make it route all local connections too like others virtual network adapters??

Another usefull thing could be command line parametters, like for retrieving my IP (ive done a script for this already) or retrieving the IPs of my friendlist (i have a tool that i use to run my game that has an option to do custom broadcast to diferent IPs and domains reading them from a txt file, to use ur tool with the one i have i should add every IP of every friend i have in Steam) .

I had hope that ur tool could bring me and my friends a better connection to play with, not usefull for me yet but i think it could be a great alternative to the usual and kinda slow VPNs.

The thing i love the most from this project is that i dont need to install anything, i could just grab my game, this and maybe steamcmd with some scripts to play in the go.

Another doubt i have is if i can connect or not to a friends game even if they have their online status set to "hidden".

Solution:
I think the option of having rooms like in LPVPN was better, bc in theory that way it can support games without options to connect trough IP itself, at the same time its even easier for new users to understand (trying to explain something to someone trough discord and see how they mess it up constantly isnt fun, this could dumb it down enough to prevent that) while keeping the option to see the IP if needed of course.

**Comment by Slinx05** [comment]
I am writing a program to modify the broadcast to custom IPs as you described (first tests worked).
It would be nice if Partylan had an interface or text file to get the IPs of friends who are online.

I have written a parser for the Partylan log file, but unfortunately the log is inconsistent. Not every friend is logged with an accepted session.

**Comment by M4RCK5** [comment]
> I am writing a program to modify the broadcast to custom IPs as you described (first tests worked).

Nice, i dont have experience with C++ or networking and havent found anything close to PartyLAN, do u have a repo where i can test ur version?

**Comment by Slinx05** [comment]
@M4RCK5 I have no deep experience with C++, but some networking experience. So I implemented a tool with Python, have a look at [GameLink](https://github.com/Slinx05/gamelink) (Pre-Release). 
**Important** you will need a VPN (Tunnel) program in addition, like partylan. My tool only make it possible to run any lan game lobby over a VPN.
